### ISSUE TITLE
With marker and cluster manager, the marker info adapter window got Invalid.

### ISSUE THREAD

**Issue by ChoungYoung** [googlemaps/android-maps-utils#992]
I use the old version 0.6.2 before, when I upgrade to the version 2.2.5, I found that the marker info window not show as before.

The old version shows like this
![WechatIMG15](https://user-images.githubusercontent.com/4734003/134268979-26a6ebb6-116b-4764-b7b4-2e1e697654d4.jpeg)

and the newest version shows just like this
![WechatIMG16](https://user-images.githubusercontent.com/4734003/134269062-da93baa0-632f-41c9-8157-2b746dd26b31.jpeg)

I've try so many times, when the display got be same with old version, the clustermanager click listener will not work.
could anybody help?

**Comment by barbeau** [comment]
@ChoungYoung Thanks for the report! This sounds similar to this issue - https://github.com/googlemaps/android-maps-utils/issues/738.

Some possible solutions are presented here:
https://github.com/googlemaps/android-maps-utils/issues/738#issuecomment-639644279

tl;dr - you'll need to update your info window adapter to use the following:

```
clusterManager.getMarkerCollection().setInfoWindowAdapter(...)
```

In v1 and higher of the library to support multiple features on the map at once the click handlers and other operations must be set on the manager and collection objects within the library. This is because `GoogleMap` only supports a single listener for events, so we have to centralize callbacks within the library to manage this.

There is more guidance on migrating to v1 and higher here:
https://github.com/googlemaps/android-maps-utils#migration-guide

Could you please take a look and let me know if that solves your issue?

If not, it would be great if you could provide a working example on GitHub that reproduces the issue.

**Comment by ChoungYoung** [comment]
> @ChoungYoung Thanks for the report! This sounds similar to this issue - #738.
> 
> Some possible solutions are presented here:
> [#738 (comment)](https://github.com/googlemaps/android-maps-utils/issues/738#issuecomment-639644279)
> 
> tl;dr - you'll need to update your info window adapter to use the following:
> 
> ```
> clusterManager.getMarkerCollection().setInfoWindowAdapter(...)
> ```
> 
> In v1 and higher of the library to support multiple features on the map at once the click handlers and other operations must be set on the manager and collection objects within the library. This is because `GoogleMap` only supports a single listener for events, so we have to centralize callbacks within the library to manage this.
> 
> There is more guidance on migrating to v1 and higher here:
> https://github.com/googlemaps/android-maps-utils#migration-guide
> 
> Could you please take a look and let me know if that solves your issue?
> 
> If not, it would be great if you could provide a working example on GitHub that reproduces the issue.

Thank you for reply, I'll try it and see if it fitable with my project.

**Comment by ChoungYoung** [comment]
> @ChoungYoung Thanks for the report! This sounds similar to this issue - #738.
> 
> Some possible solutions are presented here:
> [#738 (comment)](https://github.com/googlemaps/android-maps-utils/issues/738#issuecomment-639644279)
> 
> tl;dr - you'll need to update your info window adapter to use the following:
> 
> ```
> clusterManager.getMarkerCollection().setInfoWindowAdapter(...)
> ```
> 
> In v1 and higher of the library to support multiple features on the map at once the click handlers and other operations must be set on the manager and collection objects within the library. This is because `GoogleMap` only supports a single listener for events, so we have to centralize callbacks within the library to manage this.
> 
> There is more guidance on migrating to v1 and higher here:
> https://github.com/googlemaps/android-maps-utils#migration-guide
> 
> Could you please take a look and let me know if that solves your issue?
> 
> If not, it would be great if you could provide a working example on GitHub that reproduces the issue.

Does this means I need to add cluster with bikeClusterManager?.clusterMarkerCollection.addMarker(), Actually the marker contains some information like battery state\vehicle type, When click cluster manager I do two things, first, change the display of marker, second, get infomation from network to display a dialog, Does the infowindowadapter satisfied this need.

before cluster marker click
![181632360291_ pic](https://user-images.githubusercontent.com/4734003/134443095-0726ce42-98cb-4699-98e0-198d9c899f27.jpg)

after cluster marker click
![171632360290_ pic](https://user-images.githubusercontent.com/4734003/134443131-bf0a2461-64e0-4ce4-a9c1-9abbfd05610e.jpg)

**Comment by barbeau** [comment]
>Does this means I need to add cluster with bikeClusterManager?.clusterMarkerCollection.addMarker(),

No, cluster items should be added with `clusterManager.addItem()` directly.

I'd suggest checking out the demo app to see how the APIs should be used for v1 and higher, specifically the [CustomMarkerClusteringDemoActivity](https://github.com/googlemaps/android-maps-utils/blob/main/demo/src/gms/java/com/google/maps/android/utils/demo/CustomMarkerClusteringDemoActivity.java), and if you're using more than just the clusterManager from Android Maps Utils, then the [MultiLayerDemoActivity](https://github.com/googlemaps/android-maps-utils/blob/main/demo/src/gms/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java) too.

**Comment by ChoungYoung** [comment]
> > Does this means I need to add cluster with bikeClusterManager?.clusterMarkerCollection.addMarker(),
> 
> No, cluster items should be added with `clusterManager.addItem()` directly.
> 
> I'd suggest checking out the demo app to see how the APIs should be used for v1 and higher, specifically the [CustomMarkerClusteringDemoActivity](https://github.com/googlemaps/android-maps-utils/blob/main/demo/src/gms/java/com/google/maps/android/utils/demo/CustomMarkerClusteringDemoActivity.java), and if you're using more than just the clusterManager from Android Maps Utils, then the [MultiLayerDemoActivity](https://github.com/googlemaps/android-maps-utils/blob/main/demo/src/gms/java/com/google/maps/android/utils/demo/MultiLayerDemoActivity.java) too.

I'll try it.

**Comment by kikoso** [comment]
Hi @ChoungYoung. Did this solve your problem?

I am closing this issue, let me know otherwise.
